Modules/SubModule_Import/xASL_wrp_NII2BIDS.m

```diff
@@ -48,18 +48,28 @@
 	%% 1. Load the study parameters + dataset description
 	if ~exist(x.dir.studyPar,'file')
 		warning('Could not find the studyPar.json file...');
-		studyPar = struct;
+		studyParFull = struct;
 	else
-		studyPar = xASL_io_ReadDataPar(x.dir.studyPar, true);
+		studyParFull = xASL_io_ReadDataPar(x.dir.studyPar, true);
 	end
 	
 	% The name always has to be assigned as it is used in the DatasetDescription
-	if ~isfield(studyPar,'Name')
-		studyPar.Name = x.modules.import.imPar.studyID;
+	
+	if isfield(studyParFull,'StudyPars')
+		% For multi-context studyPar - add this to the first context
+		if ~isfield(studyParFull.StudyPars{1},'Name')
+			studyParFull.StudyPars{1}.Name = x.modules.import.imPar.studyID;
+		end
+		studyParFirstContext = studyParFull.StudyPars{1};
+	else
+		if ~isfield(studyParFull,'Name')
+			studyParFull.Name = x.modules.import.imPar.studyID;
+		end
+		studyParFirstContext = studyParFull;
 	end
 	
 	%% 2. Create the study description output and verify that all is there
-	datasetDescription = xASL_bids_CreateDatasetDescriptionTemplate(studyPar, x.Version);
+	datasetDescription = xASL_bids_CreateDatasetDescriptionTemplate(studyParFirstContext, x.Version);
```

Comment:
What if there are "more exceptions than the rule"? Why should the first be the one to use for the DataSetDescription? Probably the average of all is better...

Reply:
Average is impossible to do. The basic idea is that the first (lowest priority) entry is the one that is common for all and contains the basics, and all the others are additional things specific only in special cases. We should explain that better in the documentation. But it doesn't make sense to make an average. Worst case - we try to see if there are extra-entries relevant for datasetDescription - but the point is they shouldn't.

Reply:
This is a dangerous assumption; assuming that there is a basic protocol and that exceptions are rare. We observe in multiple studies (OASIS, NOMARED, EPAD, etc etc) that this is not the case.

So example case:
EPAD has 1600 subjects, 15 different sites. Which site is the "first" or "common" entry? (note that all this needs to be multi-sequence multi-vendor etc ready for Mathijs and ABBA)

Reply:
I don't have the answer - but this has nothing to do with my code...
The information to be written to datasetDescription has to be decided by study authors, it cannot be compiled from info inside the studyPar automatically. They can't write different datasetDescription tags to for each sequence and expect that this is merged automatically.

If you want, I can forbid to do overwriting for the datasetDescription-tags. But this has nothing to do with my JSON - that would have been the same problem for the table of course.

Besides - datasetDescription is about author, acknowledgment etc - it has nothing to do with the protocol per site.

Reply:
OK now I understand. This is solved by adding a warning for each inheritance case.